velox/exec/fuzzer/JoinFuzzer.cpp

```diff
@@ -380,6 +391,38 @@ core::JoinType JoinFuzzer::pickJoinType() {
   return kJoinTypes[idx];
 }
 
+std::string JoinFuzzer::makeJoinFilter(
+    const std::vector<std::string>& probeKeys,
+    const std::vector<std::string>& buildKeys,
+    const std::vector<RowVectorPtr>& probeInput,
+    const std::vector<RowVectorPtr>& buildInput) {
+  const auto numKeys = probeKeys.size();
+  std::string filter;
+  VELOX_CHECK_EQ(numKeys, buildKeys.size());
+  for (auto i = 0; i < numKeys; ++i) {
+    if (i > 0) {
+      filter += " AND ";
+    }
+    filter += fmt::format("{} = {}", probeKeys[i], buildKeys[i]);
+  }
+  // Add a semi filter 10% of the time.
+  if (vectorFuzzer_.coinToss(0.1) && !probeInput.empty() &&
+      !buildInput.empty()) {
+    RowTypePtr rowType = vectorFuzzer_.coinToss(0.5)
+        ? asRowType(probeInput[0]->type())
+        : asRowType(buildInput[0]->type());
+
+    for (int i = 0; i < rowType->size(); i++) {
+      // TODO: Add support for non-boolean types.
+      if (rowType->childAt(i)->isBoolean() && vectorFuzzer_.coinToss(0.5)) {
```

Comment:
Why tossing a coin at line 417 again? Also, it feels like the chance of choosing to generate a filter and rowType happens to contain a boolean column is low. What about we decide whether to generate a join filter and the filter column type in the verify() method first, and if we do, we then make build input or probe input to contain a payload column of the chosen type?